### PR TITLE
Add instructions on how to migrate packages

### PR DESCRIPTION
**Pull request by mmitche**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Review by riarenas — APPROVED** [review]

File: Documentation/MigratingPackages.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Migrating packages from nuget.org to dotnet-public
+
+The dotnet-public feed serves as .NET's OSS mirror of nuget.org, for the packages that our repos depend on. This is an alternate to the use of the 'single-feed with upstreams' approach that many projects use, due to the incompatibility of that approach with many .NET and OSS practices.
```
Comment (riarenas) [inline Documentation/MigratingPackages.md:3]: Would it be worth linking the feed here?

File: Documentation/MigratingPackages.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Microsoft-owned packages that already exist on dotnet-public are automatically updated every hour with any new versions. New package IDs, or new versions of packages **not** owned by Microsoft on nuget.org must be migrated using the following process:
+
+1. Navigate to https://dev.azure.com/dnceng/internal/_build?definitionId=931&_a=summary
+2. Queue a new build with the following parameters:
```
Comment (dougbu) [inline Documentation/MigratingPackages.md:8]:
```suggestion
2. Queue a new build with the following parameters (all are required):
```

File: Documentation/MigratingPackages.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  - Under the Migration type, select "New or non-Microsoft"
+  - Enter the package name (ID) of the package you wish to migrate
+  - Enter the package version of the package you wish to migrate.
```
Comment (dougbu) [inline Documentation/MigratingPackages.md:11]: As mentioned in your pipeline update, I suggest users will want to migrate the latest version of a package. That's mostly a nit but if we go that way, this line should mention the ability to do so. Will also need to change my earlier suggestion to something like "all but the package version are required".

**Comment by chcosta** [comment]
We should probably also update https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/882/Updating-the-dotnet-public-feed?anchor=onboarding-new-packages, at a min with a link to this doc.

**Review by chcosta — DISMISSED** [review]

File: Documentation/MigratingPackages.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Migrating packages from nuget.org to dotnet-public
```
Comment (premun) [inline Documentation/MigratingPackages.md:1]: I realize this one's quite late to the party but - is it migration when the package stays in `nuget.org`? Maybe it's mirroring?

Comment (mmitche) [reply Documentation/MigratingPackages.md]: Yep

**Review by dougbu — APPROVED** [review]
wfm :shipit:

**Pull request merged**